com.zsmartsystems.zigbee/src/main/java/com/zsmartsystems/zigbee/zcl/ZclCluster.java

```diff
@@ -1086,19 +1102,40 @@ public void removeCommandListener(final ZclCommandListener listener) {
      * Notify command listeners of an received {@link ZclCommand}.
      *
      * @param command the {@link ZclCommand} to notify
+     * @return true if at least one of the command handlers has responded to the command
      */
-    private void notifyCommandListener(final ZclCommand command) {
+    private boolean notifyCommandListener(final ZclCommand command) {
+        final AtomicBoolean response = new AtomicBoolean();
+        CountDownLatch latch = new CountDownLatch(commandListeners.size());
         for (final ZclCommandListener listener : commandListeners) {
             NotificationService.execute(new Runnable() {
                 @Override
                 public void run() {
                     logger.trace("{}: ZclCluster.notifyCommandListener {} of {}", zigbeeEndpoint.getEndpointAddress(),
                             listener, command);
 
-                    listener.commandReceived(command);
+                    if (listener.commandReceived(command)) {
+                        // A thread is responding - we don't need to wait around
+                        for (long cnt = latch.getCount(); cnt > 0; cnt--) {
+                            latch.countDown();
+                            response.set(true);
+                        }
+                        latch.notifyAll();
+                    }
+                    latch.countDown();
                 }
             });
         }
+
+        try
+
+        {
+            // TODO: Set the timer properly
```

Comment:
Out of curiosity: How do you plan to estimate a proper waiting time here?

Reply:
I plan to just set it statically (ie provide a method to allow it to be set, but not do anything to estimate it dynamically).

I'm open to other ideas though ;)